eclair-core/src/main/scala/fr/acinq/eclair/channel/ChannelData.scala

```diff
@@ -104,6 +104,7 @@ case class INPUT_INIT_CHANNEL_INITIATOR(temporaryChannelId: ByteVector32,
                                         channelType: SupportedChannelType,
                                         channelOrigin: ChannelOrigin = ChannelOrigin.Default) {
   require(!(channelType.features.contains(Features.ScidAlias) && channelFlags.announceChannel), "option_scid_alias is not compatible with public channels")
+  require(localParams.upfrontShutdownScript_opt.isDefined == Features.canUseFeature(localParams.initFeatures, remoteInit.features, Features.UpfrontShutdownScript), "upfront_shutdownscript is not consistent with local parameters")
```

Comment:
why would be remove this check ? this check should be valid and enforced for all new channels

Reply:
This is new code, so the question is why should we add it?

> this check should be valid and enforced for all new channels

True, but there are a lot of other things that we could check, but we don't, e.g. the consistency of the `dualFunded` boolean with `initFeatures`. If there is no particular reason to do it, we shouldn't. In the case of `option_scid_alias`, it makes sense because it could generate an error for the API if the parameters are not compatible with eachother. In the case of `localParams.upfrontShutdownScript_opt`, it's really an implementation detail and an internal bug, that should be verified were it is used (in `Commitments`), which it is already.

Reply:
Yes inconsistencies would be detected later but it is better to check this and fail asap than wait until we've started the opening process (which includes funding a tx etc...) ?

Reply:
Ok I see now that I misread your comment. This check is not a a spec check and should not be triggered unless we have a bug (and inconsistencies will be detected later anyway). Remove in https://github.com/ACINQ/eclair/pull/2565/commits/d94c44324622ed9f221e5e303d5aa30e08aeafe2